### ISSUE TITLE
Docker questions (and possibly suggestions)

### ISSUE THREAD

**Issue by gigafone** [onyx-platform/onyx-template#47]
I realize this is app an example to build on top of. And it is a great one at that -- well documented. In that spirit, I thought I could help make it a little better. I'm not the king of Docker, but I would like to point out some things I've seen that differ between this project's `Dockerfile` and `docker-compose.yml` from some credible resources on the web. 

Feel free to take these or leave them.
## Combine apt-get commands

From https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/:

> Always combine RUN apt-get update with apt-get install in the same RUN statement.

The current Dockerfile doesn't do this.

The current Dockerfile does the `apt-get update` on a separate line, too, but note the warning:

> Using apt-get update alone in a RUN statement causes caching issues and subsequent apt-get install instructions fail.
## Using phusion/baseimage

Note the order mentioned in https://github.com/phusion/baseimage-docker:

```
# Use phusion/baseimage as base image. To make your builds reproducible, make
# sure you lock down to a specific version, not to `latest`!
# See https://github.com/phusion/baseimage-docker/blob/master/Changelog.md for
# a list of version numbers.
FROM phusion/baseimage:<VERSION>

# Use baseimage-docker's init system.
CMD ["/sbin/my_init"]

# ...put your own build instructions here...

# Clean up APT when done.
RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

This is different from the project's Dockerfile, which has the `CMD ["/sbin/my_init"]` at the end and does not run `apt-get clean` at all.
## Pinning versions (rather than using 'latest')

In the docker-compose file:

```
  image: 'wurstmeister/zookeeper'
  # ...
  image: 'wurstmeister/kafka'
```

Would it not be better to pin to specific versions?

**Comment by lbradstreet** [comment]
These are all great points, and I agree with all of them.

I'll make the changes soon.

Thank you!

**Comment by gigafone** [comment]
@lbradstreet Cool.

Also, with regards to pinning... I took a look at some of wurstmeister's images. I'm not sure if his/her numbered builds have everything the template needs. Maybe the 'latest' are a safer bet... not sure.

**Comment by MichaelDrogalis** [comment]
Thanks for this @gigafone. It's much appreciated. :)

**Comment by gardnervickers** [comment]
Fixed with 164dcaa
